tests/Tests/ORM/Proxy/ProxyFactoryTest.php

```diff
@@ -239,6 +240,41 @@ public function testProxyClonesParentFields(): void
         self::assertSame(1000, $cloned->getSalary(), 'Expect properties on the CompanyEmployee class to be cloned');
         self::assertSame('Bob', $cloned->getName(), 'Expect properties on the CompanyPerson class to be cloned');
     }
+
+    public function testUnmanagedPropertyTriggerProxyLoad(): void
+    {
+        $product = new Product('someName');
+        $product->setImage('someImage');
+        $identifier = ['id' => 42];
+
+        $classMetaData = $this->emMock->getClassMetadata(Product::class);
+
+        $persister = $this
+            ->getMockBuilder(BasicEntityPersister::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->uowMock->setEntityPersister(Product::class, $persister);
+
+        $persister
+            ->expects(self::atLeastOnce())
+            ->method('loadById')
+            ->with(self::equalTo($identifier))
+            ->will(self::returnValue($product));
```

Comment:
Is there a difference between this and `willReturn` that you used on line 268?

Reply:
I don't think so. changing to willReturn for consistency